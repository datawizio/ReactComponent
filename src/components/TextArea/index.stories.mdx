import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import TextArea from "./index";

<Meta title="Components/Input/TextArea" component={TextArea} />

# TextArea

Компонент предназначен для вывода много строкового поля.

Базовый компонент:
[**antd/input.textarea**](https://ant.design/components/input/).

## Basic

<Preview>
  <Story name="Basic">
    <TextArea placeholder="Basic usage" rows={2} />
  </Story>
</Preview>

## Autosize

<Preview>
  <Story name="Autosize">
    <TextArea placeholder="Autosize height based on content lines" autoSize />
  </Story>
</Preview>

## Autosize with minimum and maximum

<Preview>
  <Story name="Autosize with minimum and maximum">
    <TextArea
      placeholder="Autosize with minimum and maximum"
      autoSize={{ minRows: 3, maxRows: 5 }}
    />
  </Story>
</Preview>

## With clear icon

<Preview>
  <Story name="With clear icon">
    <TextArea placeholder="With clear icon" allowClear />
  </Story>
</Preview>

## Props

- [**antd/input.textarea**](https://ant.design/components/input/#Input.TextArea)

<Props of={TextArea} />
