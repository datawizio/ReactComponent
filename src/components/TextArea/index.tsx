import React from "react";

import { Input } from "antd";
import { TextAreaProps as AntTextAreaProps } from "antd/lib/input";

import "./index.less";

export interface TextAreaProps extends AntTextAreaProps {}

const TextArea: React.FC<TextAreaProps> = props => {
  return <Input.TextArea {...props} />;
};

export default TextArea;
