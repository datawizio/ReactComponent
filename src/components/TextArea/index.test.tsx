import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";

import TextArea from "./index";

const mockProps = {};

const setUp = (props?) => mount(<TextArea {...props} />);

describe("TextArea component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("Render TextArea correctly", () => {
    expect(component).toMatchSnapshot();
  });
});
