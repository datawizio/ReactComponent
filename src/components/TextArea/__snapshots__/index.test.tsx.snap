// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea component Render TextArea correctly 1`] = `
<TextArea>
  <ForwardRef>
    <ClearableLabeledInput
      bordered={true}
      element={
        <TextArea
          className=""
          onChange={[Function]}
          prefixCls="ant-input"
        />
      }
      handleReset={[Function]}
      inputType="text"
      prefixCls="ant-input"
      value=""
    >
      <TextArea
        className=""
        onChange={[Function]}
        prefixCls="ant-input"
        value=""
      >
        <ResizableTextArea
          className=""
          onChange={[Function]}
          onKeyDown={[Function]}
          prefixCls="ant-input"
          value=""
        >
          <ResizeObserver
            disabled={true}
            onResize={[Function]}
          >
            <textarea
              className="ant-input"
              onChange={[Function]}
              onKeyDown={[Function]}
              style={Object {}}
              value=""
            />
          </ResizeObserver>
        </ResizableTextArea>
      </TextArea>
    </ClearableLabeledInput>
  </ForwardRef>
</TextArea>
`;
